### ISSUE TITLE
Error: unexpected end of JSON input while executing `supabase remote commit`

### ISSUE THREAD

**Issue by bhvngt** [supabase/cli#95]
# Bug report

## Describe the bug
I am following the `readme` instruction to setup my project. While doing so, I am getting following error while executing `supabase remote commit` command - `Error: unexpected end of JSON input`. Note that the supabase instance is brand new and I have not made any changes within the remote database.

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

1. supabase init
2. supabase remote set <supabase url>
3. supabase remote commit
4. See error - `Error: unexpected end of JSON input`

## Expected behavior

It should not throw any error and instead create the diff file between remote and local if there are changes. It should otherwise notify of no difference. 

## System information

- OS: [e.g. macOS, Windows] `macOS Intel 12.0.1`
- Version of supabase-cli - `0.15.3`
- Version of Node.js: [e.g. 10.10.0] `16.11.1`

**Comment by soedirgo** [comment]
Do you get any errors when running `supabase db remote set`? I haven't updated the default `dbVersion` so for new Supabase projects (i.e. a few days old) it needs to be changed to `140001`.

**Comment by bhvngt** [comment]
I do get an error for version mismatch while executing `supabase db remote set`. It goes away once I update the version to `140001` in the `config.json` file.

**Comment by bhvngt** [comment]
After setting up the remote, when I start supabase with `supabase start`, I am not able to connect to postgres db. I get following error - `[08P01] ERROR: pgbouncer cannot connect to server.` .  Even though the cli does not throw any error and shows up following data

```
Started local development setup.
API URL: http://localhost:54321
DB URL: postgresql://postgres:postgres@localhost:54322/postgres
Studio URL: http://localhost:54323
anon key: <anon key>
service_role key: <service_role key>
```

I also see following container list with two of the containers exiting early.

```CONTAINER ID   IMAGE                                    COMMAND                   CREATED          STATUS                      PORTS                                              NAMES
e871298afbe1   supabase/studio:latest                   "docker-entrypoint.s…"    24 seconds ago   Up 18 seconds               0.0.0.0:54323->3000/tcp                            supabase_studio_supabase-temp
5df0ea0c5b19   supabase/postgres-meta:v0.28.0           "postgres-meta"           26 seconds ago   Up 23 seconds               8080/tcp                                           supabase_pg_meta_supabase-temp
2e9f34d27e48   supabase/pgadmin-schema-diff:cli-0.0.4   "sleep infinity"          29 seconds ago   Up 25 seconds                                                                  supabase_differ_supabase-temp
39f6123ed773   supabase/storage-api:v0.9.3              "/bin/sh -c \"./stora…"   32 seconds ago   Exited (0) 19 seconds ago                                                      supabase_storage_supabase-temp
1aea7011f3d4   postgrest/postgrest:v8.0.0               "/bin/postgrest"          33 seconds ago   Up 31 seconds               3000/tcp                                           supabase_rest_supabase-temp
22295e65fc6d   supabase/realtime:v0.19.0                "./prod/rel/realtime…"    34 seconds ago   Up 32 seconds                                                                  supabase_realtime_supabase-temp
dcd7bdcf114c   inbucket/inbucket:stable                 "/start-inbucket.sh …"    35 seconds ago   Up 33 seconds (healthy)     1100/tcp, 2500/tcp, 9000/tcp                       supabase_inbucket_supabase-temp
614d8e7162a1   supabase/gotrue:v2.2.6                   "gotrue"                  36 seconds ago   Exited (1) 34 seconds ago                                                      supabase_auth_supabase-temp
55ab385121ec   kong:2.1                                 "/docker-entrypoint.…"    37 seconds ago   Up 35 seconds               8001/tcp, 8443-8444/tcp, 0.0.0.0:54321->8000/tcp   supabase_kong_supabase-temp
df5baba16975   edoburu/pgbouncer:1.15.0                 "/entrypoint.sh /usr…"    39 seconds ago   Up 37 seconds               0.0.0.0:54322->5432/tcp                            supabase_pgbouncer_supabase-temp
24cd32ebd0ae   supabase/postgres:14.1.0                 "docker-entrypoint.s…"    48 seconds ago   Up 44 seconds               5432/tcp                                           supabase_db_supabase-temp
```

Not sure if these errors are all connected. 

Thank you for looking into this issue.

**Comment by bhvngt** [comment]
@soedirgo Any update on this issue or am I doing something wrong? Right now since I am not able to connect to local postgres and with two of the containers - `gotrue` and `storage-api` crashing, the setup is almost non-functional.

**Comment by bhvngt** [comment]
no issue @soedirgo . With `v0.15.6` I executed following steps and found following results

```
mkdir supabase-temp
cd supabase-temp
supabase init
tree -L 3
```

I get following files
```
.
└── supabase
    ├── config.json
    └── migrations
```
It appears that `init.sql`, `globals.sql` and `seed.sql` are missing,

Hence, when I run `supabase db remote set`, I get `Error: supabase_migrations.schema_migrations table conflicts with the contents of supabase/migrations.`. 

Also when I start `supabase start` and then run `docker ps -a`, I see `gotrue` container exiting after the initial run.

**Comment by soedirgo** [comment]
> It appears that init.sql, globals.sql and seed.sql are missing

These are intentional, no worries there. Just updated the release notes.

> Error: supabase_migrations.schema_migrations table conflicts with the contents of supabase/migrations

I need to provide a better error message for this - basically it means you've run `supabase db remote set` with a different project. Dropping the `supabase_migrations.schema_migrations` table from the table editor should resolve this. Let me know if `supabase db remote commit` works after that.

> I see gotrue container exiting after the initial run

That's no good - can you post the `docker logs` for that?

**Comment by bhvngt** [comment]
I don't see `gotrue` exiting on my subsequent runs. So I could not reproduce this issue.

I still see `Error: unexpected end of JSON input` when I execute following steps. 

```
supabase db remote set <url>
Created a test table in the local postgres.public 
supabase db remote commit 
```
Let me know if you need any specific logs that could help you debug the issue.

**Comment by soedirgo** [comment]
Just a stab in the dark here, are you able to run `pg_dump -d $DB_URL` without errors on the remote database? If so, can you post the dump output here?

**Comment by bhvngt** [comment]
I finally found the issue. I had `@` character in my postgres password. With this I was able to set the remote db using `supabase cli`. However, `pg_dump` errored out with `could not translate host name` error. After changing the password by removing `@`, I could successfully commit to remote db.

Thanks a ton for your help. I am quiet excited to use the migration feature.

Is there a way to customise `docker containers` spawned by the `cli`. For example - customising volumes, env variables etc.
